### ISSUE TITLE
Fatal error "DDos"

### ISSUE THREAD

**Issue by ophian** [s9y/Serendipity#362]
If using the syndication plugin with rss2 only and you get a click on `/feeds/atom10.xml` (by bots or scripts) or `/feeds/atom.xml` we get some sort of DDos on our blogsite taking all resources for some time, ending with `PHP Fatal error:  Maximum execution time of 60 seconds exceeded in` Smarty compiler (smarty_internal_templatelexer.php).

We should think about an immediate fallback or exception when atom, or the rss2 is disabled in the code itself or by htaccess.

**Comment by onli** [comment]
It really depends on the settings in the syndication plugin?

**Comment by yellowled** [comment]
Shouldn't that throw a 404?

**Comment by ophian** [comment]
nned an example? :) http://yellowled.de/feeds/atom.xml

**Comment by yellowled** [comment]
Interestingly, I do not get the PHP error you mention (that would be in `error.log`, right?). At some point, the server delivers an empty XML file. (Which is not better, of course.)

**Comment by onli** [comment]
I see the error. But I can find nothing in the code explaining it. It really vanishes as soon as the syndication plugin shows the atom feed as well?

**Comment by ophian** [comment]
yes normally it is in the error.log. That is how I discovered it on a site. And it should be here too, since the source code is just blank with a 1.  

>  It really vanishes as soon as the syndication plugin shows the atom feed as well?

Did not test this case yet.

**Comment by ophian** [comment]
This seem to be two (seperate?) issues?!

It ends with a `HTTP/1.0 500 Internal Server Error` after the `max_execution` time, which is what I named to be a "DDos", since you can't get out of it and have to wait.

Response-Header Δ60185ms
X-Powered-By:	PHP/5.6.12
X-Blog:	Serendipity
Vary:	Accept-Encoding
Status:	200 OK
Server:	Apache/2.2.24 (Debian)
Pragma:	no-cache
Last-Modified:	Thu, 03 Jul 2014 08:57:32 GMT
Expires:	0
Etag:	"Thu, 03 Jul 2014 08:57:32 GMT"
Date:	Thu, 03 Sep 2015 07:16:41 GMT
Content-Type:	text/xml; charset=utf-8
Content-Length:	20
Content-Encoding:	gzip
Connection:	close
Cache-Control:	private, pre-check=0, post-check=0, max-age=0

This is what you get at the end:

> XML-Verarbeitungsfehler: Kein Element gefunden
> Adresse: http://www.example.com/feeds/atom.xml
> Zeile Nr. 1, Spalte 1:

if you allow both feeds in syndication, you don't have that (un)response time, but it fails with the same error. Is that confirmable?

For the response time issue I would love to have something immediate coded, like `set_time_limit(10);` but I assume we will have to find the other issue first.

**Comment by onli** [comment]
> if you allow both feeds in syndication, you don't have that (un)response time, but it fails with the same error. Is that confirmable?

Confirmed.

> For the response time issue I would love to have something immediate coded, like `set_time_limit(10);` but I assume we will have to find the other issue first.

Yes, we should find the real issue, stuff like set_time_limit should stay as much as possible serveradminstuff.

**Comment by ophian** [comment]
Well, the file is cached. That is why we did not have the unresponse time for a second++ request.

But I already investigated that this `Fatal Error in sysplugins/smarty_internal_templateparser.php line 1270` is a Smarty version issue matter, since Smarty 3.1.21 did work well fetching the file, apart from the other matters about atom being to strict we were talking about in other places.

I will need to inform Uwe about this, though I don't know how to make a reproducible test case for him..
